Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -58,6 +60,26 @@ static u8* s_video_buffer_pp_read_ptr;
 // polls, it's just atomic.
 // - The pp_read_ptr is the CPU preprocessing version of the read_ptr.
 
+// The next three variables are used to wakeup the GPU thread
+// - There is a fast-path if the GPU is aready running:
+//   If the atomic is set, the polling loop assure to execute the fifo.
+//   So on RunGpu, we can return immediately. In the loop however, we have to check twice.
+// - Else we fall back to the slow path. While the mutex is locked, it's stored within the
+//   atomic if the GPU waits for the condition variable.
+//   So also RunGpu must lock on notifying to fix the race condition where the GPU isn't
+//   waiting yet but the atomic is set again.
+static std::atomic<bool> s_gpu_is_running;
+static std::mutex s_fifo_mutex;
+static std::condition_variable s_fifo_cond_var;
+
+// The next two variables and "isGpuReadingData" are used to block the CPU thread until the GPU is idlying
+// The setup is similiar to the one above but without the fast path. Shuting down the GPU is rare so there is no need for it.
```

Comment:
In fact, the slower the GPU is, the better it's for us here. The slowdown @JMC47 did report is because our GPU thread is so fast that we'll wake it up for every single instruction.